functions/src/RSVP/submitRSVP.js

```diff
@@ -0,0 +1,13 @@
+const { db, functions } = require('../utils/firestore');
+
+exports.submitRSVP = functions.https.onRequest((request, response) => {
+  response.set('Access-Control-Allow-Origin', '*');
+  response.set('Access-Control-Allow-Headers', 'Content-Type, crossDomain');
+
+  const { id } = JSON.parse(request.body);
+
+  const userRef = db.collection('hacker_long_info').doc(id);
+  userRef.set(JSON.parse(request.body).rsvp, { merge: true })
```

Comment:
One other thing, can we also add a `going` tag to all RSVP applicants? Could do something like this:
```
rsvpData.tags = {}
rsvpData.tags.going = true
```

Reply:
@chamkank Okay I've added a 'going tag', put everything under an RSVP field in hacker_short_info, and added the validation step. Also, which of the other points I mentioned (ex. onBlur, error, checking to see if they've already RSVPed) do you think are most important? I'll start working on them from highest to lowest priority until we need to send out the RSVPs. 

@Ciuffi I just noticed in submitVolunteerApplication/index.js that there are a couple hacker references? Like it's using isHackerApplicationEnabled (L30, L31) and validating with constraints.hacker (L24), just wanted to make sure that this doesn't impact anything.

Reply:
Good catch I’ll check it out !

Reply:
@lynnzhang7 Looks great, thanks for the quick turnaround! I think in order of importance, it should be error (if submit fails, show some sort of error), checking to see that they've already RSVPed, and finally onBlur (tbh I don't think this last part is needed because the form doesn't have a whole lot of fields, but it would be nice to have).

Reply:
Okay sounds good! I made the check to see if they've already RSVPed, I just sent back more info in checkValidID. The submit button also now says "Submittting..." while the cloud function is running. 

For the error, do you mean doing something when submitRSVP returns status 400 here?
https://github.com/nwplus/nwhacks2019/blob/040360f8b702d7ca676ceabb5c770d6f1f988eab/web/components/rsvp/index.js#L191

Should it redirect to error 404, or I could display an error message under the submission button that says something like, "RSVP submission failed, please email hello@nwplus.io"

Reply:
Great! For the error, I think this it would mostly show up for people who lose their internet connection while filling out the RSVP form. So displaying something like "Submission failed, please try again later." without telling them to email us should be fine. We do something similar for hacker and volunteer templates as well:
https://github.com/nwplus/nwhacks2019/blob/15e47c17d12133e010eaa234fbd0d4af404bdf90/web/components/application/volunteer/pages/PageTemplate.js#L70

Reply:
Oh I see ahah that's really simple! Just implemented it.